spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1806,13 +1843,19 @@ public void handleConsumeOk(String consumerTag) {
 				latch.countDown();
 			}
 
+			@Override
+			public void handleDelivery(String consumerTag, Envelope envelope, BasicProperties properties, byte[] body)
+					throws IOException {
+				future.complete(new Delivery(envelope, properties, body));
+			}
+
 		};
 		channel.basicConsume(queueName, consumer);
 		if (!latch.await(10, TimeUnit.SECONDS)) {
```

Comment:
Do we really still need this blocking operation?
We are waiting on the `CompletableFuture<Delivery>` anyway.
And that looks more natural to wait not more than `this.receiveTimeout`.
WDYT?